File: C#/Classes/Intermedium.cs
//Esta clase es para las llantas intermedias
public class Intermedium : Tyres
{
    //Definimos los valores con este constuctor
    public Intermedium(string A, string B, string C, string D, int lapsMax, bool isWet, string tyreColor) : base(A, B, C, D, lapsMax, isWet, tyreColor)
    {
    }

    //Con este metodo se estima el clima ya que estos neumaticos son para dicho clima
    public void AreWet()
    {
        Console.WriteLine("The weather is medium");
    }

    //Sobreescribimos el metodo de la presion determinando la presion requerida para este neumatico
    public override void PSI()
    {
        Console.WriteLine("The air pressure required is 37");
    }

    //sobreescribimos el metodo durability y si lleva mas vueltas que la durabilidad maxima, llamamos al metodo de pits dentro de este
    public override string Durability(int laps)
    {
        if (laps > durabilityMax)
        {
            Pits();
            return "the tyres have been on track " + laps + " laps \nThe Intermediums are been changes";
        }
        else
        {
            return "the tyres have been on track " + laps + " laps\nCan continue";
        }
    }
}

File: Go/Intermedium.go
package main
import "fmt"

type Intermedium struct{
	Tyres
}

func NewIntermedium (A, B, C, D string, lapsMax int, isWet bool, tyreColor string) Intermedium{
	tyres := NewTyres(A, B, C, D, lapsMax, isWet, tyreColor)
	return Intermedium{Tyres: tyres}
}

func (i Intermedium) Durability(laps int) string{
	if (laps>i.durabilityMax){
		i.Pits()
		return fmt.Sprintf("The tyres have been on track %d laps\nThe Intermediums are being changed", laps)
	}else{
		return fmt.Sprintf("The tyre have been on track %d laps\nCan continue", laps)
	}
}

func (i Intermedium) PSI(){
	fmt.Println("The air pressure required is 37")
}

func (i Intermedium) AreWet(){
	fmt.Println("The wheather is medium")
}